pyproject.toml

```diff
@@ -40,6 +40,7 @@ mypy = "1.13.0"
 ruff = "^0.7.4"
 pre-commit = "^4.0.1"
 poethepoet = "^0.30.0"
+pyinstaller = { version = "^6.12.0", python = ">=3.12,<3.14" }
```

Comment:
Why a `dev` dependency instead of `cicd`?

Reply:
We keep the dev dependencies light weight because they're installed by default.

PyInstaller as part of the build & release workflow belongs in the CI/CD group for sure, if it belongs in this file at all it might be best to use latest by default with option to pin, and in which case the pin should propagate to all builders, not just Linux or macOS+Windows.

Reply:
@achidlow sounds good, i'll move it to cicd group after changes from your branch are applied